crates/client/block_import/src/verify_apply.rs

```diff
@@ -293,3 +294,651 @@ fn block_hash(
+    ///
+    /// This function generates a Header with predefined values,
+    /// useful for creating consistent test scenarios.
+    fn create_dummy_header() -> Header {
+        Header {
+            parent_block_hash: felt!("0x1"),
+            sequencer_address: felt!("0x2"),
+            block_number: 1,
+            global_state_root: felt!("0xa"),
+            transaction_count: 0,
+            transaction_commitment: Felt::ZERO,
+            event_count: 0,
+            event_commitment: Felt::ZERO,
+            state_diff_length: 0,
+            state_diff_commitment: Felt::ZERO,
+            receipt_commitment: Felt::ZERO,
+            block_timestamp: 12345,
+            protocol_version: StarknetVersion::new(0, 13, 2, 0),
+            l1_gas_price: GasPrices {
+                eth_l1_gas_price: 14,
+                strk_l1_gas_price: 15,
+                eth_l1_data_gas_price: 16,
+                strk_l1_data_gas_price: 17,
+            },
+            l1_da_mode: L1DataAvailabilityMode::Blob,
+        }
+    }
+
+    /// Creates a finalized block zero (genesis block) for testing purposes.
+    ///
+    /// This function generates a MadaraMaybePendingBlock representing the genesis block,
+    /// useful for testing scenarios involving the first block in the chain.
+    fn finalized_block_zero(header: Header) -> MadaraMaybePendingBlock {
+        let transactions = vec![
+            InvokeTransactionV1::default().into(),
+            L1HandlerTransaction::default().into(),
+            DeclareTransactionV1::default().into(),
+            DeployTransaction::default().into(),
+            DeployAccountTransactionV3::default().into(),
+        ];
+
+        let transaction_receipts = vec![
+            InvokeTransactionReceipt::default().into(),
+            L1HandlerTransactionReceipt::default().into(),
+            DeclareTransactionReceipt::default().into(),
+            DeployTransactionReceipt::default().into(),
+            DeployAccountTransactionReceipt::default().into(),
+        ];
+
+        let block_inner = MadaraBlockInner::new(transactions, transaction_receipts);
+
+        let tx_hashes = vec![Felt::from(0), Felt::from(1), Felt::from(2), Felt::from(3), Felt::from(4)];
+        let block_info = MadaraBlockInfo::new(header, tx_hashes, felt!("0x12345"));
+
+        MadaraMaybePendingBlock { info: block_info.into(), inner: block_inner }
+    }
+
+    /// Creates an empty StateDiff for testing purposes.
+    ///
+    /// This function returns a default StateDiff, useful for testing scenarios
+    /// where an empty state difference is needed.
+    fn finalized_state_diff_zero() -> StateDiff {
+        StateDiff::default()
+    }
+
+    /// Creates a dummy PreValidatedBlock for testing purposes.
+    ///
+    /// This function generates a PreValidatedBlock with predefined values,
+    /// useful for testing update_tries scenarios.
+    fn create_dummy_block() -> PreValidatedBlock {
```

Comment:
Would make sens to integrate this into #253 if possible

Reply:
it won't be possible, there we are mocking a gateway response, and we have parser to parse it from original block type to madara block type. So either we have to parse output of this function back to gateway block type.